examples/analytics/emr-on-eks/main.tf

```diff
@@ -1,62 +1,40 @@
-terraform {
-  required_version = ">= 1.0.1"
-
-  required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = ">= 3.66.0"
-    }
-    kubernetes = {
-      source  = "hashicorp/kubernetes"
-      version = ">= 2.6.1"
-    }
-    helm = {
-      source  = "hashicorp/helm"
-      version = ">= 2.4.1"
-    }
-  }
-
-  backend "local" {
-    path = "local_tf_state/terraform-main.tfstate"
-  }
+provider "aws" {
+  region = local.region
 }
 
-provider "aws" {}
-
 provider "kubernetes" {
-  experiments {
-    manifest_resource = true
+  host                   = module.eks_blueprints.eks_cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1alpha1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks_blueprints.eks_cluster_id]
   }
-  host                   = data.aws_eks_cluster.cluster.endpoint
-  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
-  token                  = data.aws_eks_cluster_auth.cluster.token
```

Comment:
I think we should keep this as data source unless we hit any specific issues.

Reply:
it is what is [recommended by the Kubernetes provider](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#exec-plugins) and also mirrors what is provided in a kubeconfig that is pulled from EKS. would there be any scenarios where we would recommend users NOT use the exec method?

Reply:
To be honest i am not sure what problem that this can introduce by using **exec plugin**.  One of the customer i worked with was using this exec plugin approach and they didn't have permissions to execute aws cli. Didn't spend time to debug further instead I have asked them to change this to data source and that worked for them.